Extension/src/Debugger/configurationProvider.ts

```diff
@@ -467,22 +480,29 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
                         }
                         const compilerDirname: string = path.dirname(compilerPath);
                         const debuggerPath: string = path.join(compilerDirname, debuggerName);
-                        if (isWindows) {
+
+                        // Check if debuggerPath exists.
+                        if (await util.checkFileExists(debuggerPath)) {
                             newConfig.miDebuggerPath = debuggerPath;
                             return resolve(newConfig);
+                        } else if ((await util.whichAsync(debuggerName)) !== undefined) {
+                            // Check if debuggerName exists on $PATH
+                            newConfig.miDebuggerPath = debuggerName;
+                            return resolve(newConfig);
                         } else {
-                            fs.stat(debuggerPath, (err, stats: fs.Stats) => {
-                                if (!err && stats && stats.isFile()) {
-                                    newConfig.miDebuggerPath = debuggerPath;
-                                } else {
-                                    newConfig.miDebuggerPath = path.join("/usr", "bin", debuggerName);
+                            if (!isWindows) {
```

Comment:
Nit: `if (!isWindows)` can be part of the `else` on line 492.

Reply:
```C++
                        } else {
                            if (!isWindows) {
                                const usrDebuggerPath: string = path.join("/usr", "bin", debuggerName);
                                if (await util.checkFileExists(usrDebuggerPath)) {
                                    newConfig.miDebuggerPath = usrDebuggerPath;
                                    return resolve(newConfig);
                                }
                            }
                            logger.getOutputChannelLogger().appendLine(localize('debugger.path.not.exists', "Unable to find the {0} debugger. The debug configuration for {2} is ignored.", `\"${debuggerName}\"`, compilerName));
                            return resolve(undefined);
                        }
```

The else has two parts, `iswindows `and `!inswindows`.
I simplify it .